pkg/tiller/resource_policy.go

```diff
@@ -34,28 +34,33 @@ const resourcePolicyAnno = "helm.sh/resource-policy"
 const keepPolicy = "keep"
 
 func filterManifestsToKeep(manifests []Manifest) ([]Manifest, []Manifest) {
-	remaining := []Manifest{}
-	keep := []Manifest{}
+	toDelete := []Manifest{}
+	toKeep := []Manifest{}
 
 	for _, m := range manifests {
 		if m.Head.Metadata == nil || m.Head.Metadata.Annotations == nil || len(m.Head.Metadata.Annotations) == 0 {
-			remaining = append(remaining, m)
+			toDelete = append(toDelete, m)
 			continue
 		}
 
 		resourcePolicyType, ok := m.Head.Metadata.Annotations[resourcePolicyAnno]
 		if !ok {
-			remaining = append(remaining, m)
+			toDelete = append(toDelete, m)
 			continue
 		}
 
 		resourcePolicyType = strings.ToLower(strings.TrimSpace(resourcePolicyType))
 		if resourcePolicyType == keepPolicy {
-			keep = append(keep, m)
+			toKeep = append(toKeep, m)
+		} else if resourcePolicyType == "" {
+			// Empty annotation means keep in Helm 2
+			toKeep = append(toKeep, m)
+		} else {
+			// Any other annotation means delete in Helm 2
+			toDelete = append(toDelete, m)
```

Comment:
I still don't think this should be accepted. The logic right in master is

1. Does the annotation `helm.sh/resource-policy` exist in the template? If yes, keep
2. Else, delete

Annotations can be added and removed from a template at any point AFAIK so why do we need to change the logic to accept `helm.sh/resource-policy: keep` to mean "delete me"? Is there some documentation somewhere or some example I can find to see why this is an issue?